src/Symfony/Bundle/FrameworkBundle/Command/RouterDebugCommand.php

```diff
@@ -109,12 +115,38 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
     private function convertController(Route $route)
     {
-        $nameParser = $this->getContainer()->get('controller_name_converter');
         if ($route->hasDefault('_controller')) {
+            $nameParser = $this->getContainer()->get('controller_name_converter');
             try {
                 $route->setDefault('_controller', $nameParser->build($route->getDefault('_controller')));
             } catch (\InvalidArgumentException $e) {
             }
         }
     }
+
+    private function convertControllerWithClassName(Route $route)
+    {
+        if (!$route->hasDefault('_controller')) {
+            return;
+        }
+
+        $controller = $route->getDefault('_controller');
+
+        if (1 === substr_count($controller, ':')) {
+            list($service, $method) = explode(':', $controller);
+            try {
+                $route->setDefault('_controller', sprintf("%s\n     method: %s::%s", $service, get_class($this->getContainer()->get($service)), $method));
```

Comment:
I'm not sure I understand what you are doing here. IIUC, you are storing the text representation of the controller in the route `_controller`? Looks like it should be done when displaying the route instead.